ts/Core/Series/Series.ts

```diff
@@ -6763,6 +6767,9 @@ class Series {
 
         series.init(chart, options);
 
+        if (casting) {
+            series.isDirtyData = true;
+        }
```

Comment:
Since this bug fix only applies to indicators that need to recalculate the data, I'm concerned that `isDirtyData` is not necessary for other series. I think this is more precise:
```ts
if (casting && series.hasDerivedData) {
    series.isDirtyData = true;
}
```

Reply:
This wont work since its the indicator that has `hasDerivedData` set, but its not the indicator that changes type and the indicator relies on the series its bound to firing `updatedData`.

Reply:
Maybe checking `linkedSeries` could work?

Reply:
Hm.. there's also antoher solution. Instead of marking series after an update with `isDirtyData` flag, technical indicators could listen to `series.update()` events and check if `type` and `data` changed in any way.

But still, it is strange that `isDirtyData` flag was not set when updating the series with `{ type, data }` object. Most of the technical indicators recalculate only on `updatedData` event (Volume By Price is recalculated on `afterSetExtremes`) .  Perhaps the best solution is to check: `casting && dataChange`?

Reply:
`isDirtyData` was only set when the series already had points previously and casting clears the points. Checking `casting && dataChange` sounds like it would break the `setData + update({ type })` case.